src/ch/fhnw/cpib/parser/Parser.java

```diff
@@ -31,43 +31,48 @@ private Token consume(Terminal expectedTerminal) throws GrammarError {
     }
 
     @Override
-    public void parse() throws GrammarError {
+    public ConcSyn.IProgram parse() throws GrammarError {
         // parsing the start symbol ...
-        /*ConcSyn.IProgram program = */program();
+        ConcSyn.IProgram program = program();
         // ... and then consuming the SENTINEL
         consume(Terminal.SENTINEL);
-        //return program;
+        return program;
     }
 
-    public void program() throws GrammarError {
+    public ConcSyn.IProgram program() throws GrammarError {
         if (token.terminal == Terminal.PROGRAM) {
             // PROGRAM IDENT <progParamList> <optGlobalCpsDecl> DO <cpsCmd> ENDPROGRAM
+            ConcSyn.Program program = new ConcSyn.Program();
             consume(Terminal.PROGRAM);
             consume(Terminal.IDENT);
```

Comment:
Müsste hier (evtl an anderen orten auch) nicht der identifier in program abgelegt werden?

Reply:
Absolut. Mir ist erst nach dem 3. oder 4. Interface aufgefallen, dass man vielleicht doch teilweise die Werte der Terminale abspeichern sollte ^^

Reply:
Ist nur das eine Mal